main.js

```diff
@@ -68,6 +70,33 @@ module.exports = function(options) {
 	var sensuChecks = sensu(name, options.sensuChecks);
 	var healthChecks = options.healthChecks;
 
+	// Note: Only *report* CSP violations.
+	// The aim is to know which scripts are being executed via third-parties.
+	if (options.withContentSecurityPolicy) {
+		csp.extend(app, {
+			reportPolicy: {
+				directives: {
+					'report-uri': ['/__csp-reports'],
+					'default-src': [
+						'unsafe-inline',
+						'unsafe-eval',
+						'self',
+						'*.ft.com',
+						'*.fyre.co',
+						'*.livefyre.com'
+					]
+				}
+			}
+		});
+
+		// Log CSP violation reports to splunk via nextLogger
+		app.get('/__csp-reports', function(req) {
```

Comment:
i think we rarely ingest from the client side…  although as these are error maybe you could use the sentry client and log them to sentry instead of splunk?